src/web/js/beforePyret.js

```diff
@@ -447,6 +447,8 @@ $(function() {
     $("#download").append(downloadElt);
   });
 
+  const CONTEXT_PREFIX = "use context ";
```

Comment:
This should be `/^use context\s+/`, so that the `.slice` on line 498 trims any interstitial whitespace between the `context` and the actual context value.

Actually, could you reuse `firstLineIsNamespace` from `beforePyret`, and expose it along with `setContextLine`, at https://github.com/brownplt/code.pyret.org/blob/ef0d4daa5c52ca28c09d659fde5b4868b95758f3/src/web/js/beforePyret.js#L294, so that you can just do `CPO.editor.firstLineIsNamespace`?  You'd fix up the regex on line 225, (it shouldn't end in `.*`, but rather `\s+`.  Then maybe refactor lines 495-498 of your code below into a function `getContextFromFirstLine()` that is likewise made available?